src/pspm_time2index.m

```diff
@@ -28,20 +28,23 @@
 end
 
 if nargin < 4
-    zero_permitted = 0;
+    is_duration = 0;
 else 
-    zero_permitted = 1;
+    is_duration = varargin{2};
 end
 
 if nargin > 4
     events = varargin{3};
     time = events(time);
 end
 
-index = double(int64(round(time * sr))); % round can sometimes result in non-integer values
+% 'round' can sometimes result in non-integer values
+index = double(int64(round(time * sr))); 
 
-if zero_permitted < 1
-    index(index == 0) = 1;
+% The first sample of the file corresponds to index 1 and time 0, i.e. we
+% assume the first sample was recorded in the interval [0, dt[ 
+if is_duration < 1
```

Comment:
This seems to be moving every element to a latter index / time point? It may become un-invertible. If this is better than changing the first element from 0 to 1, maybe keep it..